android/app-newm/src/main/java/io/newm/screens/profile/ProfileForm.kt

```diff
@@ -45,13 +47,37 @@ fun ProfileForm(
                 modifier = Modifier
                     .padding(top = 12.dp, start = 12.dp, end = 12.dp)
             ) {
-                Email(
-                    label = R.string.profile_form_nickname,
-                    emailState = nicknameState,
-                    keyboardOptions = KeyboardOptions(
-                        imeAction = ImeAction.Next,
+                if (canUserEditName) {
+                    Email(
+                        label = R.string.profile_form_first_name,
+                        emailState = firstName,
+                        keyboardOptions = KeyboardOptions(
+                            imeAction = ImeAction.Next,
+                        )
                     )
-                )
+                    Email(
```

Comment:
I will make a ticket to create an actual input component for non email fields. For now we will keep this 😓